File: 1431. Kids With the Greatest Number of Candies/Solution.c
#include <math.h>
#include <stdbool.h>
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool* kidsWithCandies(int *candies, int candiesSize, int extraCandies, int *returnSize)
{
    bool *res = (bool*)malloc(sizeof(bool) * candiesSize);
    *returnSize = candiesSize;

    int mx = 0;
    for (int i = 0; i < candiesSize; i++)
        mx = fmax(mx, candies[i]);
    for (int i = 0; i < candiesSize; i++)
        res[i] = (candies[i] + extraCandies) >= mx;
    return res;
}

File: 1431. Kids With the Greatest Number of Candies/Solution.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> res = new ArrayList<>();
        int max = Arrays.stream(candies).max().getAsInt();
        for (int candy : candies) {
            res.add((candy + extraCandies) >= max);
        }
        return res;
    }
}
